File: Homeworks/01. C# Fundamentals/04. Console-In-and-Out/06. Quadratic Equation/QuadraticEquation.cs
﻿using System;

class QuadraticEquation
{
    static void Main()
    {
        double a = double.Parse(Console.ReadLine());
        double b = double.Parse(Console.ReadLine());
        double c = double.Parse(Console.ReadLine());
        double D = Math.Pow(b, 2) - 4 * a * c;

        if (D < 0)
        {
            Console.WriteLine("no real roots");
        }

        else if (D == 0)
        {
            double x = -b / (2 * a);
            Console.WriteLine("{0:0.00}", x);
        }

        else
        {
            double x1 = (-b + Math.Sqrt(D)) / (2 * a);
            double x2 = (-b - Math.Sqrt(D)) / (2 * a);

            if (x1 < x2)
            {
                Console.WriteLine("{0:F2}", x1);
                Console.WriteLine("{0:F2}", x2);
            }

            else
            {
                Console.WriteLine("{0:F2}", x2);
                Console.WriteLine("{0:F2}", x1);
            }
        }

    }
}

File: Homeworks/06. JS Fundamentals/05. Conditional Statements/06. Quadratic equation/QuadraticEquation.js
function QuadricEquation(args) {
    var a = +args[0], b = +args[1], c = +args[2], x1 = 0, x2 = 0;

    var D = b * b - 4 * a * c;
    if (D < 0) {
        return 'no real roots';
    } else if (D == 0) {
        x1 = ((b) + Math.sqrt(D)) / (2 * a);
        console.log('x1=x2=' + x1.toFixed(2));
    }
    else {
        x1 = ((b) + Math.sqrt(D)) / (2 * a);
        x2 = ((b) - Math.sqrt(D)) / (2 * a);
        console.log('x1=' + Math.min(x1, x2).toFixed(2) + '; x2=' + Math.max(x1, x2).toFixed(2));
    }
}